src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -404,82 +455,82 @@ protected override void PostStop()
-                 * listen.AddressPromise.
-                 * */
-                .With<Listen>(listen => Listens.ContinueWith<INoSerializationVerificationNeeded>(listens =>
+            /*
+            * the first command the EndpointManager receives.
+            * instructs the EndpointManager to fire off its "Listens" command, which starts
+            * up all inbound transports and binds them to specific addresses via configuration.
+            * those results will then be piped back to Remoting, who waits for the results of
+            * listen.AddressPromise.
+            * */
+            Receive<Listen>(listen => Listens.ContinueWith<INoSerializationVerificationNeeded>(listens =>
+            {
+                if (listens.IsFaulted)
                 {
-                    if (listens.IsFaulted)
-                    {
-                        return new ListensFailure(listen.AddressesPromise, listens.Exception);
-                    }
-                    else
-                    {
-                        return new ListensResult(listen.AddressesPromise, listens.Result);
-                    }
-                }, TaskContinuationOptions.ExecuteSynchronously)
-                    .PipeTo(Self))
-                .With<ListensResult>(listens =>
+                    return new ListensFailure(listen.AddressesPromise, listens.Exception);
+                }
+                else
                 {
-                    _transportMapping = (from mapping in listens.Results
-                                        group mapping by mapping.Item1.Address
-                                            into g
-                                            select new { address = g.Key, transports = g.ToList() }).Select(x =>
-                        {
-                            if (x.transports.Count > 1)
-                            {
-                                throw new RemoteTransportException(
-                                    string.Format("There are more than one transports listening on local address {0}",
-                                        x.address));
-                            }
-                            return new KeyValuePair<Address, AkkaProtocolTransport>(x.address,
-                                x.transports.Head().Item1.ProtocolTransport);
-                        }).ToDictionary(x => x.Key, v => v.Value);
-
-                    //Register a listener to each transport and collect mapping to addresses
-                    var transportsAndAddresses = listens.Results.Select(x =>
-                    {
-                        x.Item2.SetResult(new ActorAssociationEventListener(Self));
-                        return x.Item1;
-                    }).ToList();
+                    return new ListensResult(listen.AddressesPromise, listens.Result);
+                }
+            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)
```

Comment:
> doesnt TaskScheduler.Default mean that the continuation will be scheduled on the default thread pool?

Yes, possibly. Many of these tasks are already running on the default `ThreadPool` unfortunately - this is because we don't have the same functionality that Scala does to force the TPL to use the underlying `Dispatcher` the actor is currently running on to schedule the futures.